chain/chain/src/flat_storage_resharder.rs

```diff
@@ -381,11 +393,15 @@ impl FlatStorageResharder {
                     store_update.set_flat_storage_status(
                         child_shard,
                         FlatStorageStatus::Resharding(FlatStorageReshardingStatus::CatchingUp(
-                            flat_head.hash,
+                            block_hash,
                         )),
                     );
+                    self.scheduler.send(ReshardingRequest::FlatStorageShardCatchup {
```

Comment:
Is this going from the resharding actor to itself?

Reply:
It becomes really really hard to figure out which of these functions are called from within resharding actor and which aren't. That's why I wanted us to keep all functions executed in the actor as part of the actor instead of here.

Reply:
Even just for this review, I had to bounce around several times to see who is calling who and when does a resharding request get converted to a function call in flat_storage_resharder

Reply:
Yeah now that you mentioned it I also struggled to grasp what's happening where. Do you think this should be addressed in this PR or as a follow up? @shreyan-gupta @Trisfald